driver/src/main/java/org/neo4j/driver/v1/GraphDatabase.java

```diff
@@ -168,6 +169,8 @@ public static Driver driver( URI uri, AuthToken authToken, Config config )
         {
         case "bolt":
             return new DirectDriver( address, connectionSettings, securityPlan, poolSettings, config.logging() );
+        case "bolt+discovery":
```

Comment:
The plus sign is explicitly mentioned in [RFC3986 §3.1](https://tools.ietf.org/html/rfc3986#section-3.1) as a valid character within the scheme part of a URI. Often this is used as a "link" between two related scheme parts and read as "with" or "over" such as [`git+https`](https://pip.pypa.io/en/stable/user_guide/#requirements-files).

I think here it reads more clearly than trying to combine two terms into a single word.